airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java

```diff
@@ -77,25 +78,37 @@ public static AirbyteMessageConsumer create(JdbcDatabase database,
         sqlOperations::isValidData);
   }
 
-  private static List<WriteConfig> createWriteConfigs(NamingConventionTransformer namingResolver, JsonNode config, ConfiguredAirbyteCatalog catalog) {
-    Preconditions.checkState(config.has("schema"), "jdbc destinations must specify a schema.");
+  private static List<WriteConfig> createWriteConfigs(NamingConventionTransformer namingResolver, JsonNode config, ConfiguredAirbyteCatalog catalog, boolean schemaRequired) {
+    if (schemaRequired) {
+      Preconditions.checkState(config.has("schema"), "jdbc destinations must specify a schema.");
+    }
     final Instant now = Instant.now();
-    return catalog.getStreams().stream().map(toWriteConfig(namingResolver, config, now)).collect(Collectors.toList());
+    return catalog.getStreams().stream().map(toWriteConfig(namingResolver, config, now, schemaRequired)).collect(Collectors.toList());
   }
 
-  private static Function<ConfiguredAirbyteStream, WriteConfig> toWriteConfig(NamingConventionTransformer namingResolver,
-                                                                              JsonNode config,
-                                                                              Instant now) {
+  private static Function<ConfiguredAirbyteStream, WriteConfig> toWriteConfig(
+      NamingConventionTransformer namingResolver,
+      JsonNode config,
+      Instant now,
+      boolean schemaRequired) {
     return stream -> {
       Preconditions.checkNotNull(stream.getDestinationSyncMode(), "Undefined destination sync mode");
       final AirbyteStream abStream = stream.getStream();
 
-      final String defaultSchemaName = namingResolver.getIdentifier(config.get("schema").asText());
+      final String defaultSchemaName = schemaRequired ? namingResolver.getIdentifier(config.get("schema").asText()) : namingResolver.getIdentifier(config.get("database").asText());
       final String outputSchema = getOutputSchema(abStream, defaultSchemaName);
 
       final String streamName = abStream.getName();
       final String tableName = Names.concatQuotedNames("_airbyte_raw_", namingResolver.getIdentifier(streamName));
-      final String tmpTableName = Names.concatQuotedNames("_airbyte_" + now.toEpochMilli() + "_", tableName);
+      String tmpTableName = Names.concatQuotedNames("_airbyte_" + now.toEpochMilli() + "_", tableName);
+
+      //This is for MySQL destination, the table names cant have more than 64 characters.
+      if (tmpTableName.length() > 64) {
```

Comment:
This needs collision handling in case the 0-31 and 32-63 are the same for different tables.

Reply:
Actually @ChristopheDuong mentioned that this is a more general problem. Beyond the scope of this PR.

Reply:
Yeap. Discussed this with Subodh. Will create a follow up ticket to track this.

Reply:
yes, a related issue on Postgres is here:
https://github.com/airbytehq/airbyte/issues/2948

We'd probably need to handle these table name truncations and collisions in a common class, maybe in `StandardNameTransformer` as each destination defines some character limits anyway

Reply:
Cool, Subodh, can you leave a TODO here and reference the issue? 

 e.g.
```
TODO (#2948): Refactor into StandardNameTransformed 
```
or something of the sort so we remember when we get to this.